packages/react-components/src/components/chart/store/contextDataStreams.ts

```diff
@@ -0,0 +1,39 @@
+import { DataStream } from '@iot-app-kit/core';
+import { StateCreator } from 'zustand';
+
+export interface DataStreamsData {
+  highlightedDataStreams: DataStream[];
+  hiddenDataStreams: DataStream[];
+}
+
+export interface DataStreamsState extends DataStreamsData {
+  highlightDataStream: (datastream?: DataStream) => void;
+  unHighlightDataStream: (datastream?: DataStream) => void;
+  hideDataStream: (datastream?: DataStream) => void;
+  unHideDataStream: (datastream?: DataStream) => void;
+}
+
+export const createDataStreamsSlice: StateCreator<DataStreamsState> = (set) => ({
+  highlightedDataStreams: [],
+  hiddenDataStreams: [],
+  highlightDataStream: (datastream?: DataStream) =>
```

Comment:
curious: is there a scenario where dataStream is undefined and this action needs to be called ?

Reply:
When we filter the query based on hidden or not, there are times that the query passed into get highlighted datastreams is undefined. Its also a good check for if there are no datastreams.